_posts/2019-06-27-list-strategy.md

```diff
@@ -0,0 +1,129 @@
+`--strategy=Mnemonic=` and `--strategy_regexp=Regex=`. The flags take a single strategy name as a value.  
+Example: `--spawn_strategy=linux-sandbox`
+
+- Bazel has a hardcoded and undocumented list of mnemonics that are known to work well with certain strategies and uses these 
+unless the user overrides them *individually* using `--strategy=Mnemonic=` flags.  
+Example: Bazel will use some kind of sandboxing by default if its available on your system, otherwise non-sandboxed execution (e.g. on Windows). However, it will run Javac actions via the persistent worker strategy.
+
+- Each strategy has custom code to deal with the situation that it can't execute a given action.  
+Example: the remote strategy silently falls back to sandboxed execution if an action can't run remotely. 
+The sandbox strategy silently falls back to non-sandboxed local execution if an action doesn't support sandboxing. 
+There's no good way to configure this behavior.
+
+**Example:** I want to configure my build to run remotely and fallback onto `local` strategy in case remote execution is not possible. My configuration would be something like:
+
+``` 
+$ bazel build
+ --spawn_strategy=remote
+ --strategy=Javac=remote
+ --strategy=Closure=remote
+ --strategy=Genrule=remote
+ --remote_local_fallback_strategy=local
+ ... 
+```
+
+
+## How it is after bazel 0.27
+- Bazel now *auto-detects* the execution strategy, if no strategy flag provided.  
+If none of the strategy flags was used, bazel will generate a default list of strategies `remote,worker,sandboxed,local` and, for every action it wants to execute, will pick up the first strategy that can execute it.
+
+- The user can pass comma-separated lists of strategies to the above mentioned flags: `--spawn_strategy=remote,worker,linux-sandbox`.  
+Each strategy now knows whether it can execute a given action.
+For any action that it wants to execute, Bazel just picks the first strategy from the given list (or from the default list) that claims to be able to execute the action. 
+
+- If action cannot be executed with any of the given strategies, build will fail.  
+If any of the strategy flags is provided, then bazel will use ONLY those strategies that are listed there. 
+Which means that `--spawn_strategy=remote` will NOT fallback to a `sandboxed` strategy or to any other strategy in case remote action failed or not possible at all.
+Moreover, `--remote_local_fallback_strategy=local` is now deprecated and even if it was provided, bazel will ignore it and build would still fail. 
+
+**For example**, if you try to execute `Genrule` rule remotely using `--spawn_strategy=remote`, you will get the following error:
+
+```
+ERROR: No usable spawn strategy found for spawn with mnemonic Genrule. Are your --spawn_strategy or --strategy flags too strict?
+```
+
+To fix it use a bazel *default* strategy config (remove strategy flags) or `--spawn_strategy=remote,local` or `--strategy=Gengule=local` and the build will be successful.
```

Comment:
Typo: Gengule 

Maybe explain the difference?
e.g., the first would allow local execution for non-Genrule actions as well
The second would force ALL Genrule executions to be local, whether or not marked no-remote
another possibility is "--spawn_strategy=remote --strategy=Genrule=remote,local" to ensure that:
 - all non-genrule actions are remote
 - genrule are executed remotely if possible, and locally otherwise